src/pages/index.js

```diff
@@ -1,67 +1,263 @@
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+      });
+  },
+});
+
+avatarEditPopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// Card Section Instance
+// -------------------------------------------------------------------------------------
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (cardData) => {
+      const card = new Card(
+        cardData,
+        cardTemplate,
+        (cardData) => imagePopup.open(cardData),
+        handleTrashClick,
+        updateLike
+      ).getView();
+      if (cardData.isLiked) {
+        card
+          .querySelector(".card__like-button")
+          .classList.add("card__like-button_active");
+      }
+      cardSection.appendItem(card);
+    },
+  },
+  ".cards__list"
+);
+
+// -------------------------------------------------------------------------------------
+// Initial Card Creation & Rendering from API
+// -------------------------------------------------------------------------------------
+api
+  .getCardList()
+  .then((cardList) => {
+    cardSection.renderItems(cardList);
+  })
+  .catch((error) => {
+    console.error("Failed to fetch or render cards:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Additional Card adding to API & DOM
+// -------------------------------------------------------------------------------------
 const addCardPopup = new PopupWithForm("#add-card-modal", {
   handleFormSubmit: (formData) => {
     const cardData = {
       name: formData["title"],
       link: formData["Image URL"],
     };
-    const card = createCard(cardData);
-    cardSection.prependItem(card);
-    addCardPopup.getForm().reset();
-    formValidators[
-      addCardPopup.getForm().getAttribute("name")
-    ].toggleButtonState();
-    addCardPopup.close();
+
+    addCardPopup.toggleUploadIndicator(true);
+
+    api
+      .addCard(cardData)
+      .then((apiCardData) => {
+        const card = new Card(
+          apiCardData,
+          cardTemplate,
+          (data) => imagePopup.open(data),
+          handleTrashClick,
+          updateLike
+        ).getView();
+
+        cardSection.prependItem(card);
+        addCardPopup.close();
+      })
+      .catch((error) => console.error("Error adding card:", error))
+      .finally(() => addCardPopup.toggleUploadIndicator(false));
   },
 });
+
 addCardPopup.setEventListeners();
 
-const imagePopup = new PopupWithImage("#image-modal");
-imagePopup.setEventListeners();
 // -------------------------------------------------------------------------------------
-//                               Validation
+// Card Liking
+// -------------------------------------------------------------------------------------
+
+function updateLike(card) {
+  const isCurrentlyLiked = card._likeButton.classList.contains(
```

Comment:
`_likeButton`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.